rust_toolchain/lib/dependabot/rust_toolchain/file_parser.rb

```diff
@@ -0,0 +1,108 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "sorbet-runtime"
+require "toml-rb"
+
+require "dependabot/dependency"
+require "dependabot/errors"
+require "dependabot/file_parsers"
+require "dependabot/file_parsers/base"
+require "dependabot/file_parsers/base/dependency_set"
+
+require "dependabot/rust_toolchain/package_manager"
+
+module Dependabot
+  module RustToolchain
+    class FileParser < Dependabot::FileParsers::Base
+      extend T::Sig
+
+      sig { override.returns(T::Array[Dependabot::Dependency]) }
+      def parse
+        dependency_set = DependencySet.new
+
+        dependency_files.each do |dependency_file|
+          dependency = parse_dependency_file(dependency_file)
+          next unless dependency
+
+          dependency_set << dependency
+        end
+
+        dependency_set.dependencies
+      end
+
+      sig { returns(Dependabot::Ecosystem) }
+      def ecosystem
+        @ecosystem ||= T.let(
+          Dependabot::Ecosystem.new(
+            name: "rust_toolchain",
+            package_manager: package_manager
+          ),
+          T.nilable(Dependabot::Ecosystem)
+        )
+      end
+
+      private
+
+      sig { override.void }
+      def check_required_files
+        return if dependency_files.any?
+
+        raise "No dependency files!"
```

Comment:
Nit: I suggest reporting the file name or pattern you expected to find to help anybody that hits this correct the issue in their repo.